### PR TITLE
Implement voting

### LINKED ISSUE #47: Implement MsgVote
* MsgVote
As a member of a group with an active proposal I want to submit my vote in order to support or prevent the proposal from being executed.

needed-by #31

### PR DESCRIPTION
**Pull request by alpe**
After #55 

With every vote the status of the proposal is updated. In case the decision policy returns a final result the proposal result is set and it can be executed via `MsgExec` before the timeout.

Resolves #47

### CODE REVIEW

File: incubator/group/keeper_test.go
Comment (aaronc) [inline incubator/group/keeper_test.go]: These statuses should probably be named `Proposal_Submitted` without the `Base`. Doesn't need to be addressed in this PR, just noting.

File: incubator/group/keeper_test.go
Comment (aaronc) [inline incubator/group/keeper_test.go:521]: So we're not allowing people to change votes in this model?

File: incubator/group/keeper_test.go
Comment (aaronc) [inline incubator/group/keeper_test.go:539]: In my opinion, the group version should only change when membership changes, not comment or admin because those don't affect vote tallying. So this should pass, but changing membership or policy should fail. Do you see an argument for comment or admin invalidating proposals?

File: incubator/group/types.go
Comment (aaronc) [inline incubator/group/types.go]:
```suggestion
```
Duplicate code

**Review by aaronc — CHANGES_REQUESTED** [review]

File: incubator/group/keeper_test.go
Comment (alpe) [reply incubator/group/keeper_test.go:521]: No, we had talked about this in the call with Ethan. The current model is: not changing votes and a timeout for proposal execution. With this we can run the proposal earlier.

File: incubator/group/keeper_test.go
Comment (alpe) [reply incubator/group/keeper_test.go:539]: For me this is an optimization. Simplest thing was to have any modification to upgrade the group version. Let's have a ticket to address this later. Another approach may be to keep historical versions so that modifications do not matter.

File: incubator/group/keeper_test.go
Comment (alpe) [reply incubator/group/keeper_test.go]: I modified them in this PR. Please see the commit

File: incubator/group/keeper_test.go
Comment (aaronc) [reply incubator/group/keeper_test.go:521]: Okay, so for different decision policies votes could be changed then. I think I recall now.

File: incubator/group/keeper_test.go
Comment (aaronc) [reply incubator/group/keeper_test.go:539]: Can you create another ticket to track this? I would prefer `Comment` and `Admin` don't invalidate proposals.

I think historical versions are more challenging with this ORM approach which allows for large group sizes.

**Review by aaronc — APPROVED** [review]
Let's make sure we create an issue to track proposals not being invalidated when `Comment` or `Admin` change. Otherwise LGTM.

File: incubator/group/keeper_test.go
Comment (alpe) [reply incubator/group/keeper_test.go:539]: Opened https://github.com/regen-network/cosmos-modules/issues/65

**Comment by alpe** [comment]
Waiting for approval on #55 to merge

**Pull request merged**
